graylog2-server/src/main/java/org/graylog2/lookup/TrustedPathChecker.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.lookup;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Set;
+
+@Singleton
+public class TrustedPathChecker {
+
+    private final Set<Path> trustedPaths;
+
+    @Inject
+    public TrustedPathChecker(@Named("trusted_data_file_paths") Set<Path> trustedPaths) {
+        this.trustedPaths = trustedPaths;
+    }
+
+    public boolean fileIsInTrustedPath(String filePath) throws IOException {
+        if (trustedPaths.isEmpty()) {
+            return true;
+        }
+
+        // Get the absolute file path (resolve all relative paths and symbolic links).
+        // The path.toFile().getCanonicalFile() calls accomplishes this.
+        final Path csvFilePath = Paths.get(filePath).toFile().getCanonicalFile().toPath();
```

Comment:
This has still csv in it. How about
```suggestion
        final Path absoluteFilePath = Paths.get(filePath).toFile().getCanonicalFile().toPath();
```

Reply:
We can also just use `Paths.get(filePath).toAbsolutePath().normaliize()` or `Paths.get(filePath).toRealPath()`. The latter one is throwing an exception if the path doesn't exist.

Either of these avoids the conversion to a `File` object and back.

Reply:
Sounds good. I'd avoid throwing an exception on missing files or paths here, though.
That's not the responsibility of this check

Reply:
That's a much better approach. I've changed to this.

Reply:
> We can also just use `Paths.get(filePath).toAbsolutePath().normaliize()` or `Paths.get(filePath).toRealPath()`. The latter one is throwing an exception if the path doesn't exist.
> 
> Either of these avoids the conversion to a `File` object and back.

Sorry, I've spoke to soon. This solution does not resolve symlinks, which is bad for a security related check.
I suggest we do this:

```
diff --git graylog2-server/src/main/java/org/graylog2/lookup/AllowedAuxiliaryPathChecker.java graylog2-server/src/main/java/org/graylog2/lookup/AllowedAuxiliaryPathChecker.java
index abfca75cda..110169defe 100644
--- graylog2-server/src/main/java/org/graylog2/lookup/AllowedAuxiliaryPathChecker.java
+++ graylog2-server/src/main/java/org/graylog2/lookup/AllowedAuxiliaryPathChecker.java
@@ -16,15 +16,20 @@
  */
 package org.graylog2.lookup;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import javax.inject.Inject;
 import javax.inject.Named;
 import javax.inject.Singleton;
+import java.io.IOException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.SortedSet;
 
 @Singleton
 public class AllowedAuxiliaryPathChecker {
+    private static final Logger LOG = LoggerFactory.getLogger(AllowedAuxiliaryPathChecker.class);
 
     private final SortedSet<Path> allowedPaths;
 
@@ -40,7 +45,13 @@ public class AllowedAuxiliaryPathChecker {
 
         // Get the absolute file path (resolve all relative paths and symbolic links).
         // The path.toFile().getCanonicalFile() calls accomplishes this.
-        final Path absoluteFilePath = Paths.get(filePath).toAbsolutePath().normalize();
+        final Path absoluteFilePath;
+        try {
+            absoluteFilePath = Paths.get(filePath).toAbsolutePath().toRealPath().normalize();
+        } catch (IOException e) {
+            LOG.warn("Could not check allowed path. Returning false.", e);
+            return false;
+        }
         for (Path allowedPath : allowedPaths) {
             final Path absoluteAllowedPath = allowedPath.toAbsolutePath().normalize();
             if (absoluteFilePath.startsWith(absoluteAllowedPath)) {

```